pkgs/applications/science/logic/protege/default.nix

```diff
@@ -0,0 +1,62 @@
+{ lib
+, stdenv
+, fetchzip
+, jre8
+, makeWrapper
+, makeDesktopItem
+, desktop-file-utils
+, iconConvTools
+}:
+
+stdenv.mkDerivation rec {
+  pname = "protege";
+  version = "5.5.0";
+
+  src = fetchzip {
+    url = "https://github.com/protegeproject/protege-distribution/releases/download/v${version}/Protege-${version}-platform-independent.zip";
+    sha256 = "1v82ph1pqvnc1qynhiapzw0jwm9rphsb580lc96zwsvhrr0wd690";
+  };
+
+  nativeBuildInputs = [ makeWrapper desktop-file-utils iconConvTools ];
+
+  desktopItem = makeDesktopItem {
+    name = pname;
+    exec = pname;
+    desktopName = "Protege";
+    genericName = meta.description;
+    comment = meta.description;
+    categories = "Development;";
+    icon = pname;
+    extraEntries = ''
+      StartupWMClass=${pname}
+    '';
+  };
+
+  # we need to patch the run.sh located in the extracted sources, note the deliberate whitespace!
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/src
+    mv * $out/src
+
+    substituteInPlace $out/src/run.sh \
+      --replace 'java ' "${jre8}/bin/java "
+    mkdir -p $out/bin
+    makeWrapper $out/src/run.sh $out/bin/${pname}
+
+    mkdir -p $out/share
+    ${desktopItem.buildCommand}
+    icoFileToHiColorTheme $out/src/app/Protege.ico $pname $out
+
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "Ontology editor and framework for building intelligent systems";
+    license = lib.licenses.bsd2;
```

Comment:
Protege itself is released under a 2-clause BSD, but this is the distribution version, which includes third-party plugins. These come with a variety of licenses that can be extracted from the individual JARs.

From what I could see, there are pieces that come under Apache 2.0, Eclipse Public License 1.0 and LGPL 3.0 (don't know if in the "only" or "or later" form, though). It may be good to include these additional licenses here.

Reply:
The official github protege distribution license is BSD-2 https://github.com/protegeproject/protege-distribution .
Putting other licenses in the meta field makes it sound like the user may choose which license terms they accept, which is not the case here.

Reply:
According to the relevant [manual section](https://nixos.org/manual/nixpkgs/stable/#var-meta-license), a list is used to indicate that different parts of the package are licensed differently. I agree that this can be confusing, but the solution would be to improve the meta-attribute themselves.

Also, as far as I understand the license field is used by binary caches to determine if a package can be cached and redistributed. If one of the licenses is "unfree", then I suppose the whole package would be marked "uncacheable" as to comply to licensing regulations.